### PR TITLE
unify method checks wording in docs

### PR DESCRIPTION
**Pull request by caneleex**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [ ] Library interface (affecting end-user code) 
- [x] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

this PR changes some docs to follow one style of checks.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/interactions/callbacks/IAutoCompleteCallback.java [hunk 1]
```diff
@@ -46,14 +46,13 @@ public interface IAutoCompleteCallback extends Interaction
...
-     *         If any of the following is true:
      *         <ul>
-     *             <li>{@code null} is provided</li>
-     *             <li>More than {@value OptionData#MAX_CHOICES} are added</li>
-     *             <li>Any of the choice names is empty or longer than {@value OptionData#MAX_CHOICE_NAME_LENGTH}</li>
-     *             <li>The option type is incompatible with the choice type</li>
-     *             <li>The numeric value of any of the choices is not between {@value  OptionData#MIN_NEGATIVE_NUMBER} and {@value OptionData#MAX_POSITIVE_NUMBER}</li>
-     *             <li>The string value of any of the choices is empty or longer than {@value OptionData#MAX_CHOICE_VALUE_LENGTH}</li>
+     *             <li>If {@code null} is provided</li>
+     *             <li>If more than {@value OptionData#MAX_CHOICES} are added</li>
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/interactions/callbacks/IAutoCompleteCallback.java:51]:
```suggestion
     *             <li>If more than {@value OptionData#MAX_CHOICES} choices are added</li>
```
Not really related to this PR, but thought I'd review it regardless.
Personally find 
> If more than 25 choices are added

better than
> If more than 25 are added

File: src/main/java/net/dv8tion/jda/api/interactions/callbacks/IAutoCompleteCallback.java [hunk 2]
```diff
@@ -46,14 +46,13 @@ public interface IAutoCompleteCallback extends Interaction
...
+     *             <li>If any of the choice names is empty or longer than {@value OptionData#MAX_CHOICE_NAME_LENGTH}</li>
+     *             <li>If the option type is incompatible with the choice type</li>
+     *             <li>If the numeric value of any of the choices is not between {@value  OptionData#MIN_NEGATIVE_NUMBER} and {@value OptionData#MAX_POSITIVE_NUMBER}</li>
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/interactions/callbacks/IAutoCompleteCallback.java:54]:
```suggestion
     *             <li>If the numeric value of any of the choices is not between {@value OptionData#MIN_NEGATIVE_NUMBER} and {@value OptionData#MAX_POSITIVE_NUMBER}</li>
```

File: src/main/java/net/dv8tion/jda/api/interactions/callbacks/IAutoCompleteCallback.java [hunk 3]
```diff
@@ -46,14 +46,13 @@ public interface IAutoCompleteCallback extends Interaction
...
      *         <ul>
-     *             <li>{@code null} is provided</li>
-     *             <li>More than {@value OptionData#MAX_CHOICES} are added</li>
-     *             <li>Any of the choice names is empty or longer than {@value OptionData#MAX_CHOICE_NAME_LENGTH}</li>
-     *             <li>The option type is incompatible with the choice type</li>
-     *             <li>The numeric value of any of the choices is not between {@value  OptionData#MIN_NEGATIVE_NUMBER} and {@value OptionData#MAX_POSITIVE_NUMBER}</li>
-     *             <li>The string value of any of the choices is empty or longer than {@value OptionData#MAX_CHOICE_VALUE_LENGTH}</li>
+     *             <li>If {@code null} is provided</li>
+     *             <li>If more than {@value OptionData#MAX_CHOICES} are added</li>
+     *             <li>If any of the choice names is empty or longer than {@value OptionData#MAX_CHOICE_NAME_LENGTH}</li>
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/interactions/callbacks/IAutoCompleteCallback.java:52]:
```suggestion
     *             <li>If any of the choice names are empty or longer than {@value OptionData#MAX_CHOICE_NAME_LENGTH}</li>
```

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/interactions/callbacks/IAutoCompleteCallback.java:52]: This applies to many docs line

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java [hunk 4]
```diff
@@ -448,10 +443,9 @@ public OptionData setChannelTypes(@Nonnull Collection<ChannelType> channelTypes)
      * @param  value
      *         The minimal value which can be provided for this option.
      * @throws IllegalArgumentException
-     *         If any of the following checks fail
      *         <ul>
-     *             <li>{@link OptionType type of this option} is {@link OptionType#INTEGER INTEGER} or {@link OptionType#NUMBER NUMBER}</li>
-     *             <li>{@code value} is larger than or equal to {@link OptionData#MIN_NEGATIVE_NUMBER MIN_NEGATIVE_NUMBER}</li>
+     *             <li>If {@link OptionType type of this option} is not {@link OptionType#INTEGER INTEGER} or {@link OptionType#NUMBER NUMBER}</li>
+     *             <li>If {@code value} is lower than or not equal to {@link OptionData#MIN_NEGATIVE_NUMBER MIN_NEGATIVE_NUMBER}</li>
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:448]:
```suggestion
     *             <li>If {@code value} is lower than {@link OptionData#MIN_NEGATIVE_NUMBER MIN_NEGATIVE_NUMBER}</li>
```

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:448]: applies to many

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:448]: Also, we need to pick a consistent terminology across all these docs
Options
 1) Greater than vs Less than
 2) Higher vs Lower
 3) Larger vs Smaller

Lets not mix the two

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java [hunk 5]
```diff
@@ -607,13 +596,12 @@ public OptionData setRequiredRange(double minValue, double maxValue)
...
      *
      * @throws IllegalArgumentException
-     *         If any of the following checks fail
      *         <ul>
-     *             <li>{@code name} is not null, empty and less or equal to {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
-     *             <li>{@code value} is not less than {@link #MIN_NEGATIVE_NUMBER} and not larger than {@link #MAX_POSITIVE_NUMBER}</li>
-     *             <li>The amount of already set choices is less than {@link #MAX_CHOICES}</li>
-     *             <li>The {@link OptionType} is {@link OptionType#NUMBER}</li>
-     *             <li>The option is not auto-complete enabled</li>
+     *             <li>If {@code name} is null, empty or larger than or not equal to {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:600]:
```suggestion
     *             <li>If {@code name} is null, empty, or greater than {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
```

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:600]: Applies to many

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java [hunk 6]
```diff
@@ -644,13 +632,12 @@ public OptionData addChoice(@Nonnull String name, double value)
...
-     *         If any of the following checks fail
      *         <ul>
-     *             <li>{@code name} is not null, empty and less or equal to {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
-     *             <li>{@code value} is not less than {@link #MIN_NEGATIVE_NUMBER} and not larger than {@link #MAX_POSITIVE_NUMBER}</li>
-     *             <li>The amount of already set choices is less than {@link #MAX_CHOICES}</li>
-     *             <li>The {@link OptionType} is {@link OptionType#INTEGER}</li>
-     *             <li>The option is not auto-complete enabled</li>
+     *             <li>If {@code name} is null, empty or larger than or not equal to {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
+     *             <li>If {@code value} is less than {@link #MIN_NEGATIVE_NUMBER} or larger than {@link #MAX_POSITIVE_NUMBER}</li>
+     *             <li>If the amount of already set choices is less than {@link #MAX_CHOICES}</li>
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:638]:
```suggestion
     *             <li>If adding this choice would exceed the {@link #MAX_CHOICES} allowed</li>
```

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java [hunk 7]
```diff
@@ -607,13 +596,12 @@ public OptionData setRequiredRange(double minValue, double maxValue)
...
      * @throws IllegalArgumentException
-     *         If any of the following checks fail
      *         <ul>
-     *             <li>{@code name} is not null, empty and less or equal to {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
-     *             <li>{@code value} is not less than {@link #MIN_NEGATIVE_NUMBER} and not larger than {@link #MAX_POSITIVE_NUMBER}</li>
-     *             <li>The amount of already set choices is less than {@link #MAX_CHOICES}</li>
-     *             <li>The {@link OptionType} is {@link OptionType#NUMBER}</li>
-     *             <li>The option is not auto-complete enabled</li>
+     *             <li>If {@code name} is null, empty or larger than or not equal to {@value #MAX_CHOICE_NAME_LENGTH} characters long</li>
+     *             <li>If {@code value} is less than {@link #MIN_NEGATIVE_NUMBER} or larger than {@link #MAX_POSITIVE_NUMBER}</li>
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:601]:
```suggestion
     *             <li>If {@code value} is less than {@link #MIN_NEGATIVE_NUMBER} or greater than {@link #MAX_POSITIVE_NUMBER}</li>
```

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java (hunk 6 above)
Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:638]: or maybe 
```suggestion
     *             <li>If the amount of already set choices is equal to {@link #MAX_CHOICES}</li>
```

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:638]: applies to many

File: src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java [hunk 8]
```diff
@@ -760,13 +745,12 @@ else if (type == OptionType.NUMBER)
...
-     *         If any of the following checks fail
      *         <ul>
-     *             <li>The {@link OptionType} does {@link OptionType#canSupportChoices() support choices}</li>
-     *             <li>The provided {@code choices} are not null</li>
-     *             <li>The amount of {@code choices} provided is smaller than {@link #MAX_CHOICES} when combined with already set choices</li>
-     *             <li>The {@link OptionType} of the choices is either {@link OptionType#INTEGER}, {@link OptionType#STRING} or {@link OptionType#NUMBER}</li>
-     *             <li>The option is not auto-complete enabled</li>
+     *             <li>If the {@link OptionType} does not {@link OptionType#canSupportChoices() support choices}</li>
+     *             <li>If the provided {@code choices} are null</li>
+     *             <li>If the amount of {@code choices} provided is larger than or not equal to {@link #MAX_CHOICES} when combined with already set choices</li>
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:751]:
```suggestion
     *             <li>If adding the provided choices would exceed the {@link #MAX_CHOICES} allowed</li>
```
or maybe
```suggestion
     *             <li>If the amount of {@code choices} provided, when combined with the already set choices, is greater than {@link #MAX_CHOICES}</li>
```

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java:751]: Applies to many

**Comment by caneleex** [comment]
I'm open to leave this PR/change to someone else as I won't have time for the next 2-3 weeks.

**Comment by DV8FromTheWorld** [comment]
Bumping so this doesn't get lost.

**Review by DV8FromTheWorld — APPROVED** [review]

**Pull request merged**